src/http_server.cc

```diff
@@ -3055,11 +3055,30 @@ HTTPAPIServer::StartTrace(
     TRITONSERVER_InferenceTrace** triton_trace)
 {
 #ifdef TRITON_ENABLE_TRACING
+  TraceStartOptions start_options;
+  InferenceTraceMode mode = TRACE_MODE_TRITON;
+  trace_manager_->GetTraceMode(model_name, &mode);
+  if (mode == TRACE_MODE_OPENTELEMETRY) {
+#ifndef _WIN32
+    const HttpTextMapCarrier carrier(req->headers_in);
+    auto prop =
+        otel_cntxt::propagation::GlobalTextMapPropagator::GetGlobalPropagator();
+    auto ctxt = opentelemetry::context::Context();
+    start_options.propagated_context = prop->Extract(carrier, ctxt);
```

Comment:
`propagated_context` will be set to an valid default if there is no context from client?

Reply:
Good question. Here we use an instance of class [`HttpTraceContext`](https://github.com/open-telemetry/opentelemetry-cpp/blob/c4f39f2be8109fd1a3e047677c09cf47954b92db/api/include/opentelemetry/trace/propagation/http_trace_context.h#L38C1-L38C23) as a global propagator. We set it [here](https://github.com/triton-inference-server/server/pull/6785/files#diff-c05667609085c993fd282ff18d0429945c44492b95c0d01fcbeec6946fda2db2R427)

[`Extract`](https://github.com/open-telemetry/opentelemetry-cpp/blob/c4f39f2be8109fd1a3e047677c09cf47954b92db/api/include/opentelemetry/trace/propagation/http_trace_context.h#L52) method will use [`ExtractImpl`](https://github.com/open-telemetry/opentelemetry-cpp/blob/c4f39f2be8109fd1a3e047677c09cf47954b92db/api/include/opentelemetry/trace/propagation/http_trace_context.h#L163), meaning our `Get`, e.g. HTTP [Get](https://github.com/triton-inference-server/server/pull/6785/files#diff-54db4fdff0f413084068211acf3ab675ae22c4fec2b02a680d9809dbeb756172R151), to extract `traceparent` and `tracestate` info from headers/metadata. In none is found, it will return invalid `SpanContext` and [this](https://github.com/open-telemetry/opentelemetry-cpp/blob/c4f39f2be8109fd1a3e047677c09cf47954b92db/api/include/opentelemetry/trace/propagation/http_trace_context.h#L56) line will create invalid span. So in the returned context there will be `invalid span` stored.

I believe the reason, why everything is working when no context is passed, is because by default `StartSpanOptions.parent` is initialized to invalid context [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/c4f39f2be8109fd1a3e047677c09cf47954b92db/api/include/opentelemetry/trace/span_startoptions.h#L62), 
So essentially, OpenTelemetry handles invalid spans as parents on their end to start a parent-less span. 

However, I will add changes to handle this behavior better.